### PR TITLE
tests: posix: common: separate posix threads_ext to standalone test

### LINKED ISSUE #81490: tests: posix: common: split posix_threads_ext tests into standalone test
**Is your enhancement proposal related to a problem? Please describe.**
Make the posix_threads_ext-related tests in `tests/posix/common` into a standalone test.

**Describe the solution you'd like**
The obvious way to group posix_threads_ext-related testsuites is by `POSIX_THREADS_EXT` [Option Group](https://pubs.opengroup.org/onlinepubs/9799919799/xrat/V4_subprofiles.html).

**Describe alternatives you've considered**
The way the testsuite has been designed historically.

It's probably fine to group the `POSIX_THREADS_EXT` tests into this suite as well.

**Additional context**
See #79454 for more info

### PR DESCRIPTION
**Pull request by Pancakem**
Fixes #81490

posix.common contains testsuites that can be separated into smaller groups of tests. This change moves pthread_attr into a singular testsuite at tests/posix/threads/ext app directory.

### CODE REVIEW

**Review by ycsin — DISMISSED** [review]

File: tests/posix/threads_ext/src/main.c [hunk 1]
```diff
@@ -446,7 +446,7 @@ ZTEST(pthread_attr, test_pthread_attr_setstacksize)
 	}
 }
 
-ZTEST(pthread_attr, test_pthread_attr_getscope)
+ZTEST(posix_threads_ext, test_pthread_attr_getscope)
```
Comment (cfriedt) [inline tests/posix/threads_ext/src/main.c:449]: I believe this is part of POSIX_THREADS_BASE and likely not POSIX_THREADS_EXT

But it does require the CONFIG_POSIX_THREAD_PRIORITY_SCHEDULING option.

File: tests/posix/threads_ext/src/main.c [hunk 2]
```diff
@@ -466,7 +466,7 @@ ZTEST(pthread_attr, test_pthread_attr_getscope)
 	zassert_equal(contentionscope, PTHREAD_SCOPE_SYSTEM);
 }
 
-ZTEST(pthread_attr, test_pthread_attr_setscope)
+ZTEST(posix_threads_ext, test_pthread_attr_setscope)
```
Comment (cfriedt) [inline tests/posix/threads_ext/src/main.c:469]: I believe this is part of POSIX_THREADS_BASE and likely not POSIX_THREADS_EXT

But it does require the CONFIG_POSIX_THREAD_PRIORITY_SCHEDULING option.

**Review by cfriedt — CHANGES_REQUESTED** [review]
Some of these should be part of POSIX_THREADS_BASE, others part of XSI_THREADS_EXT

**Comment by Pancakem** [comment]
> Some of these should be part of POSIX_THREADS_BASE, others part of XSI_THREADS_EXT

@cfriedt should the ones that belong to POSIX_THREADS_BASE be moved?

**Comment by cfriedt** [comment]
> @cfriedt should the ones that belong to POSIX_THREADS_BASE be moved?

Probably not until other tests for POSIX_THREADS are moved.

**Comment by Pancakem** [comment]
@cfriedt  I stripped down POSIX_THREADS_EXT to this. Is it faithful to what is needed?

File: tests/posix/common/src/pthread_attr.c
Comment (cfriedt) [inline tests/posix/common/src/pthread_attr.c:1]: We cannot remove this file while there are existing tests in it. We can only remove certain tests from this file until it is empty.

Currently, only the following tests can be removed from this file:

* test_pthread_attr_getguardsize
* test_pthread_attr_getguardsize

Additionally, the following tests from `tests/posix/common/src/mutex_attr.c` should be moved to `tests/posix/threads_ext/src/main.c`

* test_pthread_mutexattr_gettype
* test_pthread_mutexattr_settype

**Review by cfriedt — CHANGES_REQUESTED** [review]
Cannot delete all of `tests/posix/common/src/pthread_attr.c`

**Review by cfriedt — CHANGES_REQUESTED** [review]
Cannot delete all of `tests/posix/common/src/pthread_attr.c`

**Review by cfriedt — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Review by ycsin — APPROVED** [review]

**Pull request merged**